snapcraft/services/assertions.py

```diff
@@ -313,3 +320,18 @@ def edit_assertion(
                         ) from assertion_error
         finally:
             self._remove_temp_file(yaml_file)
+
+    def _validate_assertion(
```

Comment:
Am I missing something or does this not actually get an override anywhere?

Reply:
No, this isn't overridden in this PR.

This is part of the second commit, which is prep work for https://github.com/canonical/snapcraft/issues/5337 where a new `ValidationSetsService` would be able to override this function.